### PR TITLE
Valentines antagonists will no longer stop mulligan

### LINKED ISSUE #42762: valentines are in the antag list
>wiz dies
round still goes on because valentines

i need my antag reroll

### PR DESCRIPTION
**Pull request by tralezab**
Adds a variable for antagonists: no_mulligan. if set to TRUE, the mulligan won't consider this antag as worthy enough to stop mulligan.

## Why It's Good For The Game

Everybody is a valentines, so having valentines is essentially killing mulligan when valentines is essentially some loosened escalation rules!

also fixes #42762 kinda, the other half is completed in my other pr #42770 to reduce clutter

:cl:
fix: Valentines no longer hold up mulligan!
/:cl:

### CODE REVIEW

File: code/modules/antagonists/valentines/valentine.dm [hunk 1]
```diff
@@ -2,6 +2,7 @@
 	name = "valentine"
 	roundend_category = "valentines" //there's going to be a ton of them so put them in separate category
 	show_in_antagpanel = FALSE
+	no_mulligan = TRUE
```
Comment (4dplanner) [inline code/modules/antagonists/valentines/valentine.dm]: Isn't this the precise opposite

Comment (ExcessiveUseOfCobblestone) [reply code/modules/antagonists/valentines/valentine.dm]: lol.

The var name is kinda confusing, it should be `can_delay_round` or something

Comment (4dplanner) [reply code/modules/antagonists/valentines/valentine.dm]: Looks like a logic error as well, unless I'm just being dumb.
no_mulligan being true here -> var/no_mulligan is true -> return  0

Comment (ExcessiveUseOfCobblestone) [reply code/modules/antagonists/valentines/valentine.dm]: sounds "right"

Comment (tralezab) [reply code/modules/antagonists/valentines/valentine.dm]: no mulligan turned out better than mulligan_skip trust me, i like the default on this being FALSE

Comment (tralezab) [reply code/modules/antagonists/valentines/valentine.dm]: i'll switch it to prevent_mulligan, seems a lot clearer

**Comment by 4dplanner** [comment]
Now I get it 5: the re-deadening

**Comment by tralezab** [comment]
the logic was ass but we're good now

**Comment by ExcessiveUseOfCobblestone** [comment]
I summon webzodia the forbidden one

**Comment by SouDescolado** [comment]
I approve web-pr's that pass travis

**Comment by tralezab** [comment]
i was inspired by the 71 web edit pr but now i wonder how they will hand that dmi conflict?

File: code/game/gamemodes/game_mode.dm [hunk 2]
```diff
@@ -236,9 +236,14 @@
...
 		for(var/mob/Player in GLOB.alive_mob_list)
 			if(Player.mind && Player.stat != DEAD && !isnewplayer(Player) &&!isbrain(Player) && Player.client)
-				if(Player.mind.special_role || LAZYLEN(Player.mind.antag_datums)) //Someone's still antaging!
-					living_antag_player = Player
-					return 0
+				if(Player.mind.special_role || LAZYLEN(Player.mind.antag_datums)) //Someone's still antaging but is their antagonist datum important enough to skip mulligan?
+					var/prevent_roundtype_conversion = FALSE
+					for(var/datum/antagonist/antag_types in Player.mind.antag_datums)
+						if(antag_types.prevent_roundtype_conversion)
+							prevent_roundtype_conversion = TRUE
```
Comment (4dplanner) [inline code/game/gamemodes/game_mode.dm]: Probably doesn't matter much, but couldn't you break here for sweet performance gains?

Comment (4dplanner) [reply code/game/gamemodes/game_mode.dm]: It's also gross to call it antag_types when it's just one antag datum

Comment (tralezab) [reply code/game/gamemodes/game_mode.dm]: people can have multiple antag datums (brainwashed?), that's why the mind var is called antag_datums

Comment (tralezab) [reply code/game/gamemodes/game_mode.dm]: but yes the break is a great idea

File: code/game/gamemodes/game_mode.dm [hunk 3]
```diff
@@ -236,9 +236,15 @@
...
+							break
+					if(prevent_roundtype_conversion)//they were an important antag, they're our new mark
+						living_antag_player = Player
```
Comment (Jordie0608) [inline code/game/gamemodes/game_mode.dm]: You can shorten this by doing 

````
if(Player.mind && Player.stat != DEAD && !isnewplayer(Player) &&!isbrain(Player) && Player.client && Player.mind.special_role || LAZYLEN(Player.mind.antag_datums)) //Someone's still antaging but is their antagonist datum important enough to skip mulligan? 
	var/prevent_roundtype_conversion = FALSE
	for(var/datum/antagonist/antag_types in Player.mind.antag_datums)
		if(antag_types.prevent_roundtype_conversion)
			living_antag_player = Player //they're an important antag, they're our new mark
			return 0
````

**Review by Jordie0608 — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/game_mode.dm (hunk 3 above)
Comment (tralezab) [reply code/game/gamemodes/game_mode.dm]: will do

File: code/game/gamemodes/game_mode.dm [hunk 4]
```diff
@@ -235,10 +235,11 @@
 			return 0 //A resource saver: once we find someone who has to die for all antags to be dead, we can just keep checking them, cycling over everyone only when we lose our mark.
 
 		for(var/mob/Player in GLOB.alive_mob_list)
-			if(Player.mind && Player.stat != DEAD && !isnewplayer(Player) &&!isbrain(Player) && Player.client)
-				if(Player.mind.special_role || LAZYLEN(Player.mind.antag_datums)) //Someone's still antaging!
-					living_antag_player = Player
-					return 0
+			if(Player.mind && Player.stat != DEAD && !isnewplayer(Player) &&!isbrain(Player) && Player.client && Player.mind.special_role || LAZYLEN(Player.mind.antag_datums)) //Someone's still antagging but is their antagonist datum important enough to skip mulligan?
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/gamemodes/game_mode.dm]: I wonder why this checks new player, I didn't think /mob/dead was put into this list.

Comment (tralezab) [reply code/game/gamemodes/game_mode.dm]: want me to cut it?

Comment (ExcessiveUseOfCobblestone) [reply code/game/gamemodes/game_mode.dm]: I never bothered to check but it's not the only spot that does this so let's assume this is ok, can fix later lol

**Pull request merged**